saleor/graphql/attribute/mutations.py

```diff
@@ -747,14 +748,31 @@ def success_response(cls, instance):
 
     @classmethod
     def post_save_action(cls, info, instance, cleaned_input):
-        variants = product_models.ProductVariant.objects.filter(
-            Exists(instance.variantassignments.filter(variant_id=OuterRef("id")))
-        )
-
-        product_models.Product.objects.filter(
-            Q(Exists(instance.productassignments.filter(product_id=OuterRef("id"))))
-            | Q(Exists(variants.filter(product_id=OuterRef("id"))))
-        ).update(search_index_dirty=True)
+        with transaction.atomic():
+            variants = product_models.ProductVariant.objects.filter(
+                Exists(instance.variantassignments.filter(variant_id=OuterRef("id")))
+            )
+            # select_for_update needs to lock objects in consistent way to avoid
+            # deadlocks notice .order_by("pk") which ensures that.
```

Comment:
```suggestion
            # SELECT … FOR UPDATE needs to lock rows in a consistent order
            # to avoid deadlocks between updates touching the same rows.
```